### PR TITLE
Implement Elliptic Curve Crypto Signatures

### PR DESCRIPTION
**Pull request by jtdowney**
The JSON Web Algorithms draft specifies three algorithms for signatures using the Elliptic Curve Digital Signature Algorithm (ECDSA). This pull request adds support for:

* ES256 - ECDSA using P-256 and SHA-256
* ES384 - ECDSA using P-384 and SHA-384
* ES512 - ECDSA using P-521 and SHA-512

### CODE REVIEW

**Comment by excpt** [comment]
Thank you very much!

Code review in progress. :)

File: lib/jwt.rb [hunk 1]
```diff
@@ -37,10 +45,22 @@ def sign_rsa(algorithm, msg, private_key)
     private_key.sign(OpenSSL::Digest.new(algorithm.sub('RS', 'sha')), msg)
   end
 
+  def sign_ecdsa(algorithm, msg, private_key)
+    raise IncorrectAlgorithm unless NAMED_CURVES[algorithm] == private_key.group.curve_name
```
Comment (jtdowney) [inline lib/jwt.rb]: Now that I look back at this. It would probably be better to raise this with a more helpful error.

Comment (excpt) [inline lib/jwt.rb]: Feel free to extend you PR.

I can tag it as WIP if you want to. :)

Comment (jtdowney) [inline lib/jwt.rb]: Sounds good, I'll update my branch. It shouldn't take me very long.

**Comment by jtdowney** [comment]
@excpt I've pushed an update that includes the better error message

**Comment by excpt** [comment]
:+1:

**Comment by excpt** [comment]
@jtdowney Thank you for the contribution. Merged and released in version 1.5.0. :) :beers:

**Pull request merged**